test/FunctionalTests/Microsoft.OData.Core.Tests/JsonLight/ODataJsonLightValueSerializerTests.cs

```diff
@@ -216,6 +216,47 @@ public void WritingCollectionResourceValueWithPropertiesShouldWrite()
             Assert.Equal(@"[{""City"":""MyCity1""},{""City"":""MyCity2""}]", result);
         }
 
+        [Fact]
+        public void WritingResourceValueAndNestedResourceValueWithSimilarPropertiesShouldWrite()
+        {
+            var complexType = new EdmComplexType("NS", "Address");
+            complexType.AddStructuralProperty("Name", EdmPrimitiveTypeKind.String);
+            complexType.AddStructuralProperty("City", EdmPrimitiveTypeKind.String);
+            model.AddElement(complexType);
+
+            var entityType = new EdmEntityType("NS", "Customer");
+            entityType.AddStructuralProperty("Name", EdmPrimitiveTypeKind.String);
+            entityType.AddStructuralProperty("Location", new EdmComplexTypeReference(complexType, false));
+            model.AddElement(entityType);
+
+            settings.ShouldIncludeAnnotation = ODataUtils.CreateAnnotationFilter("*");
+            var result = this.SetupSerializerAndRunTest(serializer =>
+            {
+                var resourceValue = new ODataResourceValue
+                {
+                    TypeName = "NS.Customer",
+                    Properties = new[]
+                    {
+                        new ODataProperty { Name = "Name", Value = "MyName" },
+                        new ODataProperty { Name = "Location", Value = new ODataResourceValue
+                            {
+                                TypeName = "NS.Address",
+                                Properties = new [] {
+                                    new ODataProperty {  Name = "Name", Value = "AddressName" },
+                                    new ODataProperty {  Name = "City", Value = "MyCity" }
+                                }
+                            }
+                        }
+                    }
+                };
+
+                var entityTypeRef = new EdmEntityTypeReference(entityType, false);
+                serializer.WriteResourceValue(resourceValue, entityTypeRef, false, serializer.CreateDuplicatePropertyNameChecker());
+            });
+
+            Assert.Equal(@"{""Name"":""MyName"",""Location"":{""Name"":""AddressName"",""City"":""MyCity""}}", result);

```

Comment:
Could you include an example test for the failing case?

Reply:
There are several existing tests https://github.com/OData/odata.net/blob/76c691bd822fa9eaec9a54863ff50991717ce482/test/FunctionalTests/Microsoft.OData.Core.Tests/JsonLight/ODataJsonLightDeserializerTests.cs#L701
https://github.com/OData/odata.net/blob/76c691bd822fa9eaec9a54863ff50991717ce482/test/FunctionalTests/Microsoft.OData.Core.Tests/ScenarioTests/Streaming/ODataJsonLightStreamWritingTests.cs#L590